posthog/api/decide.py

```diff
@@ -519,7 +520,21 @@ def _record_feature_flag_metrics(
     ).inc()
 
     # Handle billing analytics
-    if not all(flag.startswith(SURVEY_TARGETING_FLAG_PREFIX) for flag in feature_flags.keys()):
+    # Don't charge if all the flags are survey targeting flags (both prefix and creation context)
+    should_bill = True
+
+    if feature_flags:
+        # Create a lookup for creation_context by flag key
+        creation_context_by_key = {flag.key: flag.creation_context for flag in feature_flags}
+
+        # Check if any flag should be billed (not all are survey flags with both prefix and creation context)
+        should_bill = not all(
+            flag_key.startswith(SURVEY_TARGETING_FLAG_PREFIX)
```

Comment:
Seems like we don't need to check the prefix here, do we? If I understand this correctly, if a flag didn't have the survey prefix, but somehow had `creation_context=CreationContext.Surveys`, then we'd bill them. Seems like we should rely on the new `creation_context`.